pkg/osutil/dns_darwin.go

```diff
@@ -29,12 +29,14 @@ func DNSAddresses() ([]string, error) {
 	return addresses, nil
 }
 
-func proxyURL(proxy string, port int) string {
+func proxyURL(proxy string, port interface{}) string {
 	if !strings.Contains(proxy, "://") {
 		proxy = "http://" + proxy
 	}
-	if port != 0 {
-		proxy = fmt.Sprintf("%s:%d", proxy, port)
+	if portNumber, ok := port.(float64); ok && portNumber != 0 {
```

Comment:
Is there any chance that we may get `int` too?

Also, can we print an error when we get an unexpected type?

Reply:
> Is there any chance that we may get `int` too?

No, there isn't (or shouldn't). From the `json` package docs:

>To unmarshal JSON into an interface value, Unmarshal stores one of these in the interface value:
>
> ```
> bool, for JSON booleans
> float64, for JSON numbers
> string, for JSON strings
> []interface{}, for JSON arrays
> map[string]interface{}, for JSON objects
> nil for JSON null
> ```

>Also, can we print an error when we get an unexpected type?

We could, but given the possible types, I feel like this cannot really happen. Also, the best we could do is call `logrus`.